### PR TITLE
STYLE/LINT: Set literals

### PR DESCRIPTION
**Pull request by mroeschke**
xref #22122

Use `{1, 2, 3}` instead of `set([1, 2, 3])` and enforce with a linting rule.

### CODE REVIEW

File: pandas/tests/dtypes/test_inference.py [hunk 1]
```diff
@@ -385,7 +385,7 @@ def test_convert_numeric_uint64_nan(self, coerce, arr):
 
     def test_convert_numeric_uint64_nan_values(self, coerce):
         arr = np.array([2**63, 2**63 + 1], dtype=object)
-        na_values = set([2**63])
+        na_values = {2 ** 63}
```
Comment (jbrockmendel) [inline pandas/tests/dtypes/test_inference.py]: What's the convention for whitespace around `**`?  I thought that one usually stays `2**63`.

**Comment by jbrockmendel** [comment]
Pending green, LGTM.

File: pandas/tests/dtypes/test_inference.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/dtypes/test_inference.py]: Looks like there's some ambiguity for the convention around `**` specifically: https://stackoverflow.com/questions/36794533/pep8-e226-recommendation

However this was auto-formatted automatically by PyCharm (which probably takes a strict interpretation of whitespace around arithmetic operators). I'll just convert it back to match the rest of this function (and retrigger a failed build).

**Comment by jbrockmendel** [comment]
I think we’re safe taking this off @jreback’s plate.  Merging.  Thanks @mroeschke.

**Pull request merged**